WearOAuth/oauth-device-grant/src/main/java/com/example/android/wearable/oauth/devicegrant/AuthDeviceGrantActivity.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.wearable.oauth.devicegrant
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.view.View
+import android.widget.TextView
+import androidx.activity.ComponentActivity
+import androidx.lifecycle.lifecycleScope
+import com.google.android.wearable.intent.RemoteIntent
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+/**
+ * Demonstrates the OAuth 2.0 flow on Wear OS using Device Authorization Grant, as described in
+ * [RFC 8628](https://datatracker.ietf.org/doc/html/rfc8628).
+ *
+ * This sample currently directly polls the Google OAuth 2.0 API, which means the client id and
+ * client secret are included in the app. In practice you would use an intermediary server that
+ * holds these values for you.
+ *
+ * The sample uses the Google OAuth 2.0 APIs, but can be easily extended for any other
+ * OAuth 2.0 provider.
+ *
+ * See [AuthDeviceGrantViewModel] for the implementation of the authorization flow.
+ */
+class AuthDeviceGrantActivity : ComponentActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_auth)
+        // In a real world situation you would use some form of Dependency Injection here.
+        val viewModel = AuthDeviceGrantViewModel()
```

Comment:
I think we can still use `by viewModels<AuthDeviceGrantViewModel>()` here instead, even without Hilt or DI.

Reply:
Would be possible, but with my recent changes there's an argument added to the constructor. I _could_ add a custom `ViewModelProvider.Factory`, but for the sake of this sample I don't think it's that useful. What do you think?

Reply:
Just looked into this a bit more, and I think we should definitely avoid constructing the `ViewModel` directly.

As is, the lifecycle of the `ViewModel` isn't hooked up in the normal manner, which means that nothing will clear the `ViewModel` and clean up things like the `viewModelScope`.

Also, the new usage is giving an activity reference to the `ViewModel`, which it can't have.

I think `RemoteIntentHelper` can alternatively take an application `Context` instead of an `Activity`, so `AuthDeviceGrantViewModel` can be an `AndroidViewModel` to get that reference. I'm not sure exactly what functionality that gives up though compared to the `RemoteIntentHelper` created with an `Activity`.

Reply:
Fixed!